Mail/Views/Menu Drawer/FolderListView.swift

```diff
@@ -24,15 +24,19 @@ import RealmSwift
 import SwiftUI
 
 struct NestableFolder: Identifiable {
-    var id: Int {
+    let id: Int
+    let detachedContent: Folder
+    let children: [NestableFolder]
+
+    init(content: Folder, children: [NestableFolder]) {
         // The id of a folder depends on its `remoteId` and the id of its children
         // Compute the id by doing an XOR with the id of each child
-        return children.reduce(content.remoteId.hashValue) { $0 ^ $1.id }
+        let contentToUse = content.detached()
+        id = children.reduce(contentToUse.remoteId.hashValue) { $0 ^ $1.id }
```

Comment:
_new_ Here the id is computed only once, since it relies on a `detached` object.